### PR TITLE
Avoid string to HTML conversion

### PR DESCRIPTION
**Pull request by fregante**
I'm not particularly sure if it's worth it. I had to copy [the attributes loop from `stringify-attributes`](https://github.com/sindresorhus/stringify-attributes/blob/main/index.js)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,13 +22,35 @@ const linkify = (href, options) => createHtmlElement({
 		...options.attributes,
 		href, // eslint-disable-line no-dupe-keys
 	},
-	text: typeof options.value === 'undefined' ? href : undefined,
-	html: typeof options.value === 'undefined' ? undefined
-		: (typeof options.value === 'function' ? options.value(href) : options.value),
```
Comment (fregante) [inline index.js:16]: If this PR is not merged (I think not), I'll send another PR to rewrite this

**Comment by sindresorhus** [comment]
Does this have any benefits?

**Comment by fregante** [comment]
It’s the right way to do it 🥹 

Currently the code takes an object, turns it into a string, and then creates the DOM. This feels backwards.

The original intent was to avoid innerHTML/parsing, but the current API does not allow us to drop it altogether. 

The reason to drop them is because it reduces the chance to make XSS mistakes and because some analysis software may warn if `innerHTML` is found in the code.

Since this package it’s not used much and no one else is asking for this change, I’m not really pushing for this PR to be merged.

**Comment by sindresorhus** [comment]
Alright. I'm convinced. 👍

**Comment by fregante** [comment]
Lost interest. It has a couple of drawbacks:

- it does not eliminate `innerHTML` from the source, especially if a callback function is passed
- it duplicates the logic from `create-html-element`
- the API would have to be altered altogether to give access to the DOM element rather than only textContent/innerHTML

**Pull request closed without merging**